src/decimal.rs

```diff
@@ -549,7 +552,14 @@ impl Decimal {
             }
         } else {
             Decimal { flags, hi, lo, mid }
-        }
+        };
+
+        assert!(
```

Comment:
Since this is a crate only function I'd perhaps remove this assert. Only because it can _sometimes_ be useful to generate invalid scales etc.